scripts/dev-setup/main.py

```diff
@@ -0,0 +1,138 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import argparse
+import os
+import site
+import typing
+from dataclasses import dataclass
+from pathlib import Path
+
+from test_resources import REGISTERED_TEST_RESOURCES_FUNCTIONS
+from utils import REPO_ROOT_DIR, change_cwd, print_blue, run_cmd
+
+PROMPT_FLOW_PKGS = [
+    "promptflow-tracing",
+    # TODO: uncomment below lines when the packages are ready
+    # "promptflow-core",
+    # "promptflow-devkit",
+    # "promptflow-azure",
+    "promptflow[azure]",
+]
+
+
+def set_up_git_hook_scripts(verbose: bool) -> None:
+    run_cmd(["pip", "install", "pre-commit"], verbose=False)  # ensure pre-commit is installed
+    cmd = ["pre-commit", "install"]
+    print_blue(f"Running {cmd} to set up the git hook scripts")
+    run_cmd(cmd, verbose=verbose)
+
+
+def collect_and_install_from_pyproject() -> None:
+    # collect dependencies from pyproject.toml and install
+    deps = []
+    # cwd is already changed to the package folder in the context
+    # so we can safely open pyproject.toml here
+    with open("pyproject.toml", mode="r", encoding="utf-8") as f:
+        lines = f.readlines()
+        collect_flag = False
+        for line in lines:
+            line = line.strip()
+            if line.startswith("["):
+                if "[tool.poetry.group.dev.dependencies]" in line or "[tool.poetry.group.test.dependencies]" in line:
```

Comment:
does poetry require user to do something like this?

Reply:
if user has poetry installed, we can leverage command `poetry export` to extract these deps to a req txt, so we don't need to do this brute force way; also, if user only uses poetry (without something like conda), then dev setup is very easy with simply `poetry install`, in that way, might not need a dev setup script at all.